### PR TITLE
kill the fuse process if the app container has exited long time

### PR DESCRIPTION
**Pull request by zxh326**

### CODE REVIEW

File: pkg/controller/app_controller.go [hunk 1]
```diff
@@ -113,6 +138,29 @@ func (a *AppController) umountFuseSidecar(pod *corev1.Pod, fuseContainer corev1.
...
+	}
+	return nil
+}
+
+func (a AppController) killFuseProcess(pod *corev1.Pod, fuseContainer corev1.Container) error {
+	if fuseContainer.Name == "" {
+		return nil
+	}
+	log := klog.NewKlogr().WithName("app-ctrl").WithValues("pod", pod.Name, "namespace", pod.Namespace)
+	cmd := []string{"sh", "-c", "kill 1"}
```
Comment (zwwhdls) [inline pkg/controller/app_controller.go]: the service process is not 1 in old version(without exec)

File: pkg/controller/app_controller.go [hunk 2]
```diff
@@ -113,6 +138,29 @@ func (a *AppController) umountFuseSidecar(pod *corev1.Pod, fuseContainer corev1.
...
+	}
+	return nil
+}
+
+func (a AppController) killFuseProcess(pod *corev1.Pod, fuseContainer corev1.Container) error {
+	if fuseContainer.Name == "" {
+		return nil
+	}
+	log := klog.NewKlogr().WithName("app-ctrl").WithValues("pod", pod.Name, "namespace", pod.Namespace)
+	cmd := []string{"sh", "-c", "ps -x | grep mount.juicefs | grep -v grep | awk '{print $1}' | xargs kill"}
```
Comment (timfeirg) [inline pkg/controller/app_controller.go]: parent process pid is 1? maybe use 1 instead of grep?

Comment (zxh326) [reply pkg/controller/app_controller.go]: see https://github.com/juicedata/juicefs-csi-driver/pull/1113#discussion_r1758256432

Comment (timfeirg) [reply pkg/controller/app_controller.go]: pkill can make it shorter

```
root@juicefs-demo1:~# ./juicefs mount chaos-ee-test /mnt/jfs/chaos
...OK, chaos-ee-test is ready at /mnt/jfs/chaos.
root@juicefs-demo1:~# ps -ef | grep juicefs
root       78344       1 10 11:31 ?        00:00:00 ./juicefs mount chaos-ee-test /mnt/jfs/chaos
root       78352   78344 16 11:31 ?        00:00:00 /root/.juicefs/jfsmount mount chaos-ee-test /mnt/jfs/chaos
root       78365   77218  0 11:31 pts/2    00:00:00 grep --color=auto juicefs

# parent & child are all killed
root@juicefs-demo1:~# pkill jfsmount -e
jfsmount killed (pid 78344)
jfsmount killed (pid 78352)
```

**Review by zwwhdls — APPROVED** [review]

**Pull request merged**